otoutfilter/OTOutFilter.go

```diff
@@ -96,7 +98,9 @@ func (cc *otOutFilter) readFirstLine(p []byte) int {
 				// remove the log
 				simplelogger.AssertTrue(logIdx[1] == len(firstline))
 				logStr := strings.TrimSpace(firstline)
-				cc.printLog(logStr)
+				logIdxStr := logPattern.FindStringSubmatch(firstline)
```

Comment:
Matching the regex twice seems unefficient. 

Can we call `FindStringSubmatch` or `FindStringSubmatchIndex` once?